builder/swig/python.bzl

```diff
@@ -123,38 +134,73 @@ swig_python_wrapper = rule(
 )
 
 
-def swig_python(name, lib, shared_lib_name=None, **kwargs):
+def swig_python(*, name, lib, shared_lib_name=None, python_headers, libpython, **kwargs):
     swig_wrapper_name = name + "__swig"
     if not shared_lib_name:
         shared_lib_name = "_" + name
 
     swig_python_wrapper(
-        class_name = name,
         name = swig_wrapper_name,
+        module_name = name,
         shared_lib_name = shared_lib_name,
         lib = lib,
+        python_headers = python_headers,
+        libpython = select({
+            "@vaticle_dependencies//util/platform:is_linux": libpython,
+            "@vaticle_dependencies//util/platform:is_mac": None,
+            "@vaticle_dependencies//util/platform:is_windows": libpython,
+        }),
         **kwargs,
     )
 
-    def swig_cc_binary(shared_lib_filename):
-        # name doesn't accept select()
-        native.cc_binary(
-            name = shared_lib_filename,
-            deps = [lib, swig_wrapper_name],
-            srcs = [swig_wrapper_name],
-            linkshared = True,
-        )
-
-    swig_cc_binary(shared_lib_name + ".so")
-    swig_cc_binary(shared_lib_name + ".lib")
-
-    native.alias(
+    native.cc_binary(
         name = shared_lib_name,
-        actual = select({
-            "@vaticle_dependencies//util/platform:is_mac": (shared_lib_name + ".so"),
-            "@vaticle_dependencies//util/platform:is_linux": (shared_lib_name + ".so"),
-            "@vaticle_dependencies//util/platform:is_windows": (shared_lib_name + ".lib"),
-        })
+        deps = [lib, swig_wrapper_name],
+        srcs = [swig_wrapper_name],
+        linkshared = True,
+        linkopts = select({
+            "@vaticle_dependencies//util/platform:is_windows": ["ntdll.lib"],
```

Comment:
Let's leave a comment here in the code

Reply:
`ntdll.lib` is necessary when linking in Windows, so we need to specify this `linkopts` argument for the linker.